app/controllers/global_user_roles_controller.rb

```diff
@@ -0,0 +1,75 @@
+class GlobalUserRolesController < GlobalSettingsController
+
+  before_filter :load_user, only: [:destroy, :edit, :update]
+
+  def index
+    @users = UserPresenter.wrap(User.with_global_roles)
+  end
+
+  def destroy
+    destroy_all_global_roles
+    redirect_to global_user_roles_url
+  end
+
+  def edit
+    if @user == current_user
+      flash[:error] = translate("self_not_allowed", action: "change")
+      redirect_to global_user_roles_url
+    else
+      @user = UserPresenter.new(@user)
+    end
+  end
+
+  def search
+  end
+
+  def update
+    case
+    when params[:roles].blank?
+      destroy_all_global_roles
+    when @user == current_user
```

Comment:
I'm guessing we won't get to this point because it's prevented by the `edit` rules, but should this check happen before the `params[:roles].blank?` case so a user _definitely_ can't remove their own permissions.